src/index.ts

```diff
@@ -1,16 +1,62 @@
 import { Web3PluginBase } from "web3";
+import { StarknetRpcApi } from "./types";
+import * as _methods from "./methods";
 
-export class StarknetPlugin extends Web3PluginBase {
+export class StarknetPlugin extends Web3PluginBase<StarknetRpcApi> {
   public pluginNamespace = "starknet";
 
   public test(param: string): void {
```

Comment:
remove^^

Reply:
It should stay as `requestManager` extends that type (via `Web3Context` that uses this generic)

Reply:
oh yeah and it defaults to `type Web3APISpec = Record<string, (...params: any) => any> | unknown;` if generic is not specified

Reply:
test function?!

Reply:
will be removed